Add 1D Profile viewer doc for IRIS Level 2 data sets (#26)
glue-viz/glue-solar: docs/guide-to-glue-1dprofile-viewer-for-iris-data.rst

.. _guide_to_1dprofile_viewer_for_iris_data:

=========================================================================
A Guide to using Glue's 1D Profile viewer to probe IRIS Level 2 data sets
=========================================================================

> Loading multi-scan IRIS Level 2 date sets and stacking the sequential raster cubes

Reviewer: ```suggestion
Loading multi-scan IRIS Level 2 data sets and stacking the sequential raster cubes
```

   :alt: Choosing the directory containing both raster and SJI data sets of the same observation

Then, choose the relevant raster and SJI data cubes of different scans and remember to stack
the sequential raster cubes in order to obtain 4D data cubes from 3D ones. An example view of this
step is provided below:

.. image:: images/choosing-iris-level-2-data-cubes-and-stacking-raster-cubes.png
   :width: 800
   :alt: Importing Multi-scan IRIS Level 2 data sets with the directory loader

We have chosen 3 raster and 2 SJI data sets to import as an example to illustrate this functionality.
Once the IRIS data sets have been successfully loaded, we will see the data sets showing up in the
Data Collection window in the upper left of the GUI, as follows:

.. image:: images/raster-and-sji-data-loaded-to-data-collection.png
   :width: 800
   :alt: Loaded raster and SJI data sets in the Data section of the Data Collection window

Now we are ready to visualize the 2D slices of the imported data cubes.

Using Glue's 2D Image viewer to get the indices and slices needed for the 1D spectrum plot
------------------------------------------------------------------------------------------

> To plot 2D slices of the N-D data cubes, we will need to drag the data ("C_II_1336" in this

Reviewer: ```suggestion
To plot 2D slices of the ND data cubes, we will need to drag the data ("C_II_1336" in this
```

> example) concerned from the Data Collection area and drop it at the large plotting window

Reviewer: ```suggestion
example) concerned from the "Data Collection" area and drop it at the large plotting window
```

Using Glue's 1D Profile viewer to generate 1D spectra at different spatial and temporal positions
-------------------------------------------------------------------------------------------------
Like for the 2D slices case, we will need to drag and drop the same data set onto the plotting
window in order to start the 1D Profile viewer, with the ultimate goal to generate non-collapsed
1D spectra that change with changes in the slider values. Please keep in mind to select the exact
same data and not either of the ``IndexedData`` object or the ``SlicedData`` object that shares
a very similar representation, otherwise an error would be thrown:

.. image:: images/dragging-and-dropping-the-same-data-to-viewing-window.png
   :width: 800
   :alt: Dragging and dropping the same data to the viewing window to get the 1D Profile viewer

.. image:: images/choosing-the-1d-profile-viewer.png
   :width: 800
   :alt: Choosing the 1D Profile viewer to generate 1D spectra of the sun

Once the 1D Profile viewer shows up, you will be shown a collapsible version of the ``Maximum``
function of the 1D spectrum. However, this may not be very telling and is not a genuine 1D

> spectrum, but is some statistical representation of the maximum of the quantity concerned.

Reviewer: ```suggestion
spectrum, but is a statistical representation of the maximum of the quantity concerned.
```
